bin/compare_api_surfaces.py

```
@@ -0,0 +1,147 @@
+
+import json
+import re
+import sys
+
+
+def _normalize_type(type_str: str) -> str:
+    result = type_str
+    result = re.sub(r"\bUnion\[([^,]+),\s*None\]", r"Optional[\1]", result)
+    result = re.sub(r"\bOptional\[([^]]+)\]", r"\1 | None", result)
+    result = re.sub(r"typing\.", "", result)
+    return result
```

Ruff (0.15.7)

[warning] Unnecessary assignment to `result` before `return` statement

Remove unnecessary assignment

(RET504)